fast_barcode_scanner/ios/Classes/BarcodeReader.swift

```diff
@@ -134,32 +135,44 @@ class BarcodeReader: NSObject {
 		}) else {
 			throw ReaderError.cameraNotSuitable(arguments.resolution, arguments.framerate)
 		}
-
-		try! captureDevice.lockForConfiguration()
-		captureDevice.activeFormat = optimalFormat
-		captureDevice.activeVideoMinFrameDuration = optimalFormat.videoSupportedFrameRateRanges.first!.minFrameDuration
-		captureDevice.activeVideoMaxFrameDuration = optimalFormat.videoSupportedFrameRateRanges.first!.minFrameDuration
-		captureDevice.unlockForConfiguration()
-
+        
+        do {
+            try captureDevice.lockForConfiguration()
+            captureDevice.activeFormat = optimalFormat
+            captureDevice.activeVideoMinFrameDuration = optimalFormat.videoSupportedFrameRateRanges.first!.minFrameDuration
+            captureDevice.activeVideoMaxFrameDuration = optimalFormat.videoSupportedFrameRateRanges.first!.minFrameDuration
+            captureDevice.unlockForConfiguration()
+        } catch {
+            throw ReaderError.configurationLockError(error)
+        }
+        
 		previewSize = CMVideoFormatDescriptionGetDimensions(captureDevice.activeFormat.formatDescription)
 	}
 
-	func start(fromPause: Bool) {
+	func start(fromPause: Bool) throws {
+        guard captureDevice != nil else { return }
+        
 		captureSession.startRunning()
 
 		if !fromPause {
 			self.textureId = textureRegistry.register(self)
 		}
 
 		if (torchActiveBeforeStop) {
-			try! captureDevice.lockForConfiguration()
-			captureDevice.torchMode = .on
-			captureDevice.unlockForConfiguration()
-			torchActiveBeforeStop = false
+            do {
+                try captureDevice.lockForConfiguration()
+                captureDevice.torchMode = .on
+                captureDevice.unlockForConfiguration()
+                torchActiveBeforeStop = false
+            } catch {
+                throw ReaderError.configurationLockError(error)
+            }
 		}
 	}
 
 	func stop(pause: Bool) {
+        guard captureDevice != nil else { return }
```

Comment:
This looks like it will fix the issue. Okay for me to approve the change?

Reply:
The underlying issue might not be fixed with this. But as long as everything works as expected, it should be fine